boardgame-ui/src/pages/ResetPasswordPage.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <div>
+    <section v-if="token && userId" class="section">
+      <div class="box">
+        <h1 class="title">{{$t('auth.reset-password.title')}}</h1>
+        <p v-if="error" class="error">{{$t('error.invalid-credentials')}}</p>
+
+        <form @submit.prevent="validateBeforeSubmit">
+          <b-field  password-reveal
+                    :label="$t('label.password')"
+                    :type="{'is-danger': errors.has('password')}"
+                    :message="errors.first('password')">
+            <b-input v-model="password"
+                    password-reveal type="password"
+                    v-validate="{ required: true, min: 8 }"
+                    name="password" ref="password"></b-input>
+          </b-field>
+
+          <b-field 
+                    :label="$t('label.confirmPassword')"
+                    :type="{'is-danger': errors.has('confirm-password')}"
+                    :message="errors.first('confirm-password')">
+            <b-input v-model="confirmPassword"
+                    password-reveal type="password"
+                    v-validate="'required|confirmed:password'"
+                    name="password"
+                    :disabled="!password"></b-input>
+          </b-field>
+          <p>{{$t('label.passwordHint')}}</p>
+
+          <p class="control">
+            <button class="button is-primary is-fullwidth">
+              {{$t('label.submit')}}
+            </button>
+          </p>
+        </form>
+
+      </div>
+    </section>
+
+    <section v-else>
+      <b-message type="is-danger" has-icon>
+        {{$t('auth.reset-password.no-token-message')}}
+      </b-message>
+    </section>
+    
+  </div>
+</template>
+
+<script>
+import Authentication from '@/utils/api/Authentication';
+
+export default {
+  data() {
+    return {
+      password: '',
+      confirmPassword: '',
+      error: false,
+    };
+  },
+
+  computed: {
+    token() {
+      return this.getToken();
+    },
+
+    userId() { 
+      return this.getUserId();
+    }
+  },
+
+  methods: {
+    getToken() {
+      if (!('token' in this.$route.query)) {
+        return null;
+      }
+      return this.$route.query.token;
+    }, 
+
+    getUserId() {
+      if (!('id' in this.$route.query)) {
+        return null;
+      }
+      return parseInt(this.$route.query.id, 10);
+    },
+
+    validateBeforeSubmit() {
+      this.$validator.validateAll().then(async (valid) => {
+        if (valid) {
+          try {
+            await Authentication.resetPassword(this.getToken(), this.getUserId(), this.password);
```

Comment:
```suggestion
            await Authentication.resetPassword(this.token, this.userId, this.password);
```

And here, you could use the computed properties instead of the methods (prevents unnecessary evaluation)

Reply:
hahaha, just found that out by trial and error before seeing this :p